src/messages/mod.rs

```diff
@@ -58,6 +60,67 @@ impl HopMessage {
     }
 }
 
+#[derive(Eq, PartialEq, Clone)]
+pub struct HopMessageWithSerializedMessage {
```

Comment:
Suggestions to make the names shorter without losing information:
- `HopMessageWithSerializedMessage` -> `HopMessageWithBytes`
- `serialized_message` -> `bytes`
- `new_with_serialized_message` -> `new_with_bytes`

Reply:
Yes, looks pretty good, just one thing, the bytes are not just the serialized SignedRoutingMessage, but the actual on the wire message which is SignedRoutingMessage wrapped in enum and hop message.

But if that does not seem to be confusing I'll go with the rename you suggest (not sure my names do a better job making that distinction to be honest).

Reply:
Hm, I see. Well, if you think that distinction is important to keep, them maybe keep the name.